crypto/eth2_key_derivation/src/derived_key.rs

```diff
@@ -81,9 +82,31 @@ fn derive_child_sk(parent_sk: &[u8], index: u32) -> ZeroizeHash {
 ///
 /// Equivalent to `HKDF_mod_r` in EIP-2333.
 fn hkdf_mod_r(ikm: &[u8]) -> ZeroizeHash {
-    let prk = hkdf_extract(b"BLS-SIG-KEYGEN-SALT-", ikm);
-    let okm = &hkdf_expand(prk, MOD_R_L);
-    mod_r(okm.as_bytes())
+    // ikm = ikm + I2OSP(0,1)
+    let mut ikm_with_postfix = SecretBytes::zero(ikm.len() + 1);
+    ikm_with_postfix.as_mut_bytes()[..ikm.len()].copy_from_slice(ikm);
+
+    // info = "" + I2OSP(L, 2)
+    let mut info = [0; 2];
+    BigEndian::write_int(&mut info, MOD_R_L as i64, 2);
+
+    let mut output = ZeroizeHash::zero();
+    let zero_hash = ZeroizeHash::zero();
+
+    let mut salt = b"BLS-SIG-KEYGEN-SALT-".to_vec();
+    let mut hashed_salt = [0; HASH_SIZE];
```

Comment:
I think you can do away with the `hashed_salt` variable:

```rust
    let mut salt = b"BLS-SIG-KEYGEN-SALT-".to_vec();
    while output.as_bytes() == zero_hash.as_bytes() {
        let mut hasher = Sha256::new();
        hasher.update(salt);
        salt = hasher.finalize().to_vec();

        let prk = hkdf_extract(&salt, ikm_with_postfix.as_bytes());
        let okm = &hkdf_expand(prk, &info, MOD_R_L);

        output = mod_r(okm.as_bytes());
    }
```

It looks a bit more like the spec this way, too.